x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/search/enrich_monitor_groups.ts

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, sortBy } from 'lodash';
+import { QueryContext } from '../elasticsearch_monitor_states_adapter';
+import { getHistogramInterval } from '../../../helper';
+import { INDEX_NAMES, STATES } from '../../../../../common/constants';
+import {
+  MonitorSummary,
+  SummaryHistogram,
+  Check,
+  CursorDirection,
+  SortOrder,
+} from '../../../../../common/graphql/types';
+import { MonitorEnricher } from './fetch_page';
+
+export const enrichMonitorGroups: MonitorEnricher = async (
+  queryContext: QueryContext,
+  checkGroups: string[]
+): Promise<MonitorSummary[]> => {
+  // TODO the scripted metric query here is totally unnecessary and largely
+  // redundant with the way the code works now. This could be simplified
+  // to a much simpler query + some JS processing.
+  const params = {
+    index: INDEX_NAMES.HEARTBEAT,
+    body: {
+      query: {
+        bool: {
+          filter: [{ terms: { 'monitor.check_group': checkGroups } }],
```

Comment:
I attempted to only minimally alter this query and leave the obvious simplification here for a future PR since this one is big enough already. The only that's changed is that now we filter for exactly the docs we need (which of course makes most of the scripted metric redundant)

Reply:
Yes. Or, we can eliminate this stage complete and make this a per-monitor operation which I believe @shahzad31 is doing in a separate PR.